File: loop.pl
use strict;
use warnings;

exit main();

sub main {
  printf("--------------------------\n");

  my $top = 20000;
  my $expected = $top * $top;
  printf("Perl: running up to %d (%d) iterations\n", $top, $expected);

  my $total = 0;
  for ( 0 .. $top - 1 ) {
    for ( 0 .. $top - 1 ) {
      $total += 1;
    }
  }
  printf("%s\n", $total == $expected ? "OK" : "BAD");
  return 0;
}

File: loop.js
console.log(`--------------------------`);

let top = 20000;
let expected = top * top;
console.log(`Node: running up to ${top} (${expected}) iterations`);

let total = 0;
for (let i = 0; i < top; i++) {
  for (let j = 0; j < top; j++) {
    total += 1;
  }
}
console.log(`${total == expected ? "OK" : "BAD"}`);
